Objects/dictobject.c

```diff
@@ -331,7 +350,7 @@ dictkeys_decref(PyInterpreterState *interp, PyDictKeysObject *dk)
 #ifdef Py_REF_DEBUG
     _Py_DecRefTotal(_PyInterpreterState_GET());
 #endif
-    if (--dk->dk_refcnt == 0) {
+    if (_Py_atomic_add_ssize(&dk->dk_refcnt, -1) == 1) {
```

Comment:
How about adding a macro for the free-threading version and the default version?

Reply:
I did similar approach at listobject: https://github.com/python/cpython/pull/114651/commits/393cbefe027440f41f9099da7c4915d0e7e8529f

Reply:
@corona10 is there a significant difference in performance between the two builds? I'm not sure if it is worth it to add these preprocessor guards everywhere if there is no measurable effect :) Is there a different motivation than performance?

Reply:
Regardless of whether we do the macro approach everywhere, I think it's a good idea in this specific case (specifically for performance).